EIPS/nonreentrant.md

```diff
@@ -0,0 +1,52 @@
+---
```

Comment:
```suggestion
--- 
eip: 7705
```
Assigning next sequential EIP/ERC/RIP number.

Please also update the filename.